packages/data/src/api/tokenById/tokenById.ts

```diff
@@ -0,0 +1,47 @@
+import { fetchGraphQl } from "../../graphql/fetch";
+import { errorContractAddress, errorToken } from "./tokenById.mock";
+import { tokenByIdQuery } from "./tokenbyId.query";
+
+import { TokenByIdResults } from "./tokenById.types";
+
+export const checkParams = (tokenId: string, contractAddress: string) => {
+  if (typeof tokenId !== "string") {
+    return errorToken;
+  } else if (typeof contractAddress !== "string") {
+    return errorContractAddress;
+  } else {
+    return { valid: true, message: "" };
+  }
+};
+
+export const tokenById = async (
+  tokenId: string,
+  contractAddress: string
+): Promise<TokenByIdResults> => {
+  const { valid, message } = checkParams(tokenId, contractAddress);
+
+  if (valid) {
```

Comment:
Personal preference perhaps, but to me it's much clearer to return early with validation errors rather than wrap the entire rest of the method e..g. `if (!valid) { tell the user and return }`

Reply:
ok,  will fix this by tomorrow